enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk/embed.v1.ts

```diff
@@ -0,0 +1,170 @@
+import type {
+  SdkIframeEmbedSettings,
+  SdkIframeEmbedTagMessage,
+  SdkIframeEmbedTagSettings,
+} from "./types/embed";
+
+const EMBEDDING_ROUTE = "embed/sdk/v1";
+
+type EmbedSettingKey = keyof SdkIframeEmbedSettings;
+
+const ALLOWED_EMBED_SETTING_KEYS = [
+  "apiKey",
+  "instanceUrl",
+  "dashboardId",
+  "questionId",
+  "template",
+  "theme",
+  "locale",
+] as const satisfies EmbedSettingKey[];
+
+type AllowedEmbedSettingKey = (typeof ALLOWED_EMBED_SETTING_KEYS)[number];
+
+class MetabaseEmbed {
+  static readonly VERSION = "1.0.0";
+
+  private _settings: SdkIframeEmbedTagSettings;
+  private _isEmbedReady: boolean = false;
+  private iframe: HTMLIFrameElement | null = null;
+
+  constructor(settings: SdkIframeEmbedTagSettings) {
+    if (!settings.apiKey || !settings.instanceUrl) {
```

Comment:
Why isn't this condition inside `_validateEmbedSettings`? It feels to me this belongs there.

Reply:
This was because `_validateEmbedSettings` is also used for the `embed.updateSettings()` public method where the embedder can update their settings dynamically, and therefore `updateSettings({ theme })` would fail to validate because it does not have an `apiKey` and `instanceUrl`.

But now that I think about it, we can just do `_validateEmbedSettings` inside `_setEmbedSettings` after we merged the settings and the code will be much simpler, because `apiKey` and `instanceUrl` will always be there. Let me update this!

Reply:
Fixed in https://github.com/metabase/metabase/pull/56915/commits/92ca062bfc9e07821972a4d22084a5bc9791d16f -- I moved this to `_validateEmbedSettings`